src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/SubaccountTransactionSupervisor.kt

```diff
@@ -147,6 +148,36 @@ internal class SubaccountTransactionSupervisor(
         return submitCancelOrder(orderId, marketId, callback, payload, analyticsPayload, uiClickTimeMs)
     }
 
+    internal fun cancelAllOrders(marketId: String?, callback: TransactionCallback): HumanReadableCancelAllOrdersPayload {
+        val payload = payloadProvider.cancelAllOrdersPayload(marketId)
+
+        payload.payloads.forEach { cancelPayload ->
+            val subaccount = stateMachine.state?.subaccount(subaccountNumber)
+            val existingOrder = subaccount?.orders?.firstOrNull { it.id == cancelPayload.orderId }
+                ?: throw ParsingException(
+                    ParsingErrorType.MissingRequiredData,
+                    "no existing order to be cancelled for $cancelPayload.orderId",
+                )
+            val analyticsPayload = analyticsUtils.cancelOrderAnalyticsPayload(
```

Comment:
nit: can we use named args here

Reply:
Let Android Studio do this for you:

https://github.com/user-attachments/assets/553d8728-a2d2-4e5f-8588-45b3ae6c2798